elooplib/client.cpp

```diff
@@ -35,32 +37,95 @@ int Client::handleConnection()
     return 0;
 }
 
-void Client::prepateData()
+void Client::recognizeData()
 {
-    if (response.substr(0, 5) == "time\n")
-        Client::prepareTime();
-    else if (response.substr(0, 5) == "echo ")
-        Client::prepareEcho();
+    std::string filter{" \n"};
+
+    auto pos = response.find_first_of(filter);
+    std::string command = response.substr(0, pos + 1);
+
+    if (command == "time\n")
+    {
+        Client::handleTime();
+        return;
+    }
+
+    if (command == "echo ")
+    {
+        Client::handleEcho();
+        return;
+    }
+
+    if (command == "compress ")
+    {
+        Client::handleCompress();
+        return;
+    }
     else
+    {
         response = "Wrong request\n";
+    }
 }
 
-void Client::prepareEcho()
+void Client::handleEcho()
 {
     response = response.substr(5, request_field.size() - 5);
 }
 
-void Client::prepareTime()
+void Client::handleTime()
 {
     auto t = std::chrono::system_clock::now();
     auto tt = std::chrono::system_clock::to_time_t(t);
     response = std::ctime(&tt);
 }
 
+void Client::handleCompress()
+{
+    response = response.substr(9, request_field.size() - 10);
+    std::cout << response << std::endl; // DEBUG
+    const std::filesystem::path dir(response);
+    if (!exists(dir))
+    {
+        response = "Path is not exists\n";
+        return;
+    }
+
+    int dir_len = 0;
+    // TODO try_lock if(1) etc
+    std::list<std::string> files;
+    for (const auto &entry : std::filesystem::directory_iterator(response))
+    {
+        std::string tmp = entry.path();
+        if(tmp.find(".gz") == std::string::npos)
+        {
+            //std::remove_if(begin(tmp), end(tmp), '\"'); TODO
+            files.push_back(tmp); 
+
+        }
+
+        dir_len++;
+    }
+    
+    files_counter = 0;
+    for(const auto &f: files)
+        compressFile(f);
+
+    response = std::to_string(files_counter);
```

Comment:
Currently whole event loop is blocked until every file is not compressed, you should redesign your event loop to avoid it. Take a look at std future